src/vector/mod.rs

```diff
@@ -92,18 +92,18 @@ impl Vector {
 }
 
 impl Coordinate for Vector {
-    fn magnitude(self) -> f64 
-    {
+    fn magnitude(self) -> f64 {
         self.magnitude()
     }
 
-    fn distance(self, v1: Self) -> f64
-    {
+    fn distance(self, v1: Self) -> f64 {
         self.distance(v1)
     }
 
-    fn lerp(self, v1: Self, t: f64) -> Self
-    {
+    fn lerp(self, v1: Self, t: f64) -> Self {
         self.lerp(v1, t)
     }
 }
+pub fn vector(x: f64, y: f64) -> Vector {
```

Comment:
What's the purpose of this function?

Reply:
We already have the macro `vec2!` (not added by me), I think you should edit the macro to use Vector::from_components instead and drop this function.